accounts-db/benches/bench_read_only_accounts_cache.rs

```diff
@@ -0,0 +1,208 @@
+#![feature(test)]
+
+extern crate test;
+
+use {
+    criterion::{criterion_group, criterion_main, BenchmarkId, Criterion},
+    ndarray::{Array2, ArrayView},
+    rand::{rngs::SmallRng, seq::SliceRandom, SeedableRng},
+    solana_accounts_db::{
+        accounts_db::AccountsDb, read_only_accounts_cache::ReadOnlyAccountsCache,
+    },
+    solana_sdk::{
+        account::{Account, ReadableAccount},
+        pubkey::Pubkey,
+    },
+    std::{
+        sync::{
+            atomic::{AtomicBool, Ordering},
+            Arc,
+        },
+        thread::Builder,
+    },
+};
+
+const NUM_READERS_WRITERS: &[usize] = &[
+    8,
+    16,
+    // These parameters are likely to freeze your computer, if it has less than
+    // 32 cores.
+    // 32, 64, 128, 256, 512, 1024,
+];
+
+/// Benchmarks the read-only cache eviction mechanism. It does so by performing
+/// multithreaded reads and writes on a full cache. Each write triggers
+/// eviction. Background reads add more contention.
```

Comment:
This is a great bench, bit it tests the worst case right? Do we have benches for the cases when we have no evictions, and also for when we have a "normal" amount of evictions? Normal being X every 500ms or so? Something that approximates the current eviction rate on mnb.

We want to test the worst case, and we want to improve perf with higher throughput than current mnb, but we also want to ensure we don't accidentally regress with current load.

Reply:
> This is a great bench, but it tests the worst case right? Do we have benches for the cases when we have no evictions, and also for when we have a "normal" amount of evictions? Normal being X every 500ms or so? Something that approximates the current eviction rate on mnb.

No, we don't. This is the only bench which triggers any eviction at all. I will add a case closer to the current mnb numbers.

Reply:
Well, the "no eviction" case is handled here - https://github.com/anza-xyz/agave/blob/8c3a6bb6898e61b0962d6fe9a892dd29b11da097/accounts-db/benches/accounts.rs#L126-L181

But I'm not a big fan of it, because:

- It does `iter().collect()` inside the bench.
- It adds only 1000 accounts.
- Doesn't use criterion. :stuck_out_tongue: 

So I think it makes sense to bench that case in this file.

Reply:
I don't have opinions here, up to you and accounts-db peeps to decide whether to improve this bench or consolidate with the new ones you're writing!

Reply:
I ended up adding a "no eviction" case with contention as well.